toqito/state_opt/bell_inequality_max.py

```diff
@@ -0,0 +1,151 @@
+"""Computes the upper bound for a given bipartite Bell inequality."""
+
+from itertools import combinations
+
+import cvxpy as cp
+import numpy as np
+
+from toqito.channels import partial_transpose
+from toqito.perms import permutation_operator, swap
+
+
+def bell_inequality_max(
+    joint_coe: np.ndarray,
+    a_coe: np.ndarray,
+    b_coe: np.ndarray,
+    a_val: np.ndarray,
+    b_val: np.ndarray,
+    solver_name: str = "SCS",
+) -> float:
+    r"""Return the upper bound for the maximum violation(Tsirelson Bound) for a given bipartite Bell Inequality.
+
+    This computes the upper bound for the maximum value of a given bipartite Bell Inequality using an SDP.
+    The method is from :cite:`Navascues_2014_Characterization` and the implementation is based on :cite:`QETLAB_link`.
+    This is useful for various tasks in device independent quantum information processing.
+
+    The function formulates the problem as a SDP problem in the following format
+
+    .. math::
```

Comment:
add a blankline after you call mathmode. Right now, your equations are not rendered correctly because of this. 

https://toqito--1026.org.readthedocs.build/en/1026/autoapi/state_opt/bell_inequality_max/index.html

Reply:
that's weird. ruff shouldn't be doing this. let me double check.

Reply:
I wonder if the issue is related to math mode having trouble with a multiline equation? And not the fact that there isn't a blankline between where you call mathmode and where the equation starts?

Reply:
Another multiline example that works:

https://github.com/vprusso/toqito/blob/5fff14a785fbe4a38f3e396d1898115e3b2a442a/toqito/state_metrics/fidelity_of_separability.py#L45-L61